webextension/background.js

```diff
@@ -1,3 +1,63 @@
+browser.contextMenus.create({
+  id: "open-in-this-container",
+  title: "Always Open in This Container",
+  contexts: ["all"],
+});
+
+function getUserContextIdFromCookieStore(cookieStore) {
+  const container = cookieStore.replace("firefox-container-", "");
+  if (container !== cookieStore) {
+    return container;
+  }
+  return false;
+}
+
+const storagePrefix = "originMap@@_";
+const storageArea = browser.storage.local;
+
+browser.contextMenus.onClicked.addListener((info, tab) => {
+  const pageUrl = new window.URL(info.pageUrl);
+  const userContextId = getUserContextIdFromCookieStore(tab.cookieStoreId);
+  // Mapping ${pageUrl.host} to ${userContextId}
+  if (userContextId) {
+    const originStoreKey = `${storagePrefix}${pageUrl.host}`;
+    storageArea.set({[originStoreKey]: userContextId});
+  }
+});
+
+browser.webRequest.onBeforeRequest.addListener((options) => {
```

Comment:
Could we also add this check?
```
if (options.frameId !== 0) {
  return {}
}
```
So we only do this logic for the requests that happen in the top frame?

I.e., we just want to check the first/top request in the tab - not all the requests?